src/community/graticule/src/main/java/org/geotools/data/graticule/GraticuleDataStore.java

```diff
@@ -0,0 +1,387 @@
+import org.geotools.data.graticule.gridsupport.LineFeatureBuilder;
+import org.geotools.feature.NameImpl;
+import org.geotools.feature.simple.SimpleFeatureTypeBuilder;
+import org.geotools.geometry.jts.ReferencedEnvelope;
+import org.locationtech.jts.geom.LineString;
+
+public class GraticuleDataStore implements DataStore {
+    static final Logger log = Logger.getLogger("GraticuleDataStore");
+    final ReferencedEnvelope bounds;
+    ArrayList<Double> steps;
+    final Name name;
+
+    private static GraticuleHandler handler;
+    public final SimpleFeatureType schema;
+
+    public GraticuleDataStore(ReferencedEnvelope env, List<Double> steps) {
+        this.steps = new ArrayList<>(steps);
+        this.bounds = env;
+        Collections.sort(steps);
+        StringBuilder n = new StringBuilder("Graticule(");
+        n.append(steps.get(0));
+        if (steps.size() > 1) {
+            n.append("-").append(steps.get(steps.size() - 1));
+        }
+        n.append(")");
+        this.name = cleanName(new NameImpl(n.toString()));
+        schema = buildType(n.toString(), bounds.getCoordinateReferenceSystem());
+    }
+
+    /**
+     * Information about this service.
+     *
+     * <p>This method offers access to a summary of header or metadata information describing the
+     * service. Subclasses may return a specific ServiceInfo instance that has additional
+     * information (such as FilterCapabilities).
+     *
+     * @return SeviceInfo
+     */
+    @Override
+    public ServiceInfo getInfo() {
+
+        DefaultServiceInfo info = new DefaultServiceInfo();
+        info.setDescription("Features from " + getClass().getSimpleName());
+        try {
+            info.setSchema(new URI(name.getNamespaceURI()));
+        } catch (URISyntaxException e) {
+            throw new RuntimeException(e);
+        }
+        return info;
+    }
+
+    /**
+     * Creates storage for a new <code>featureType</code>.
+     *
+     * <p>The provided <code>featureType</code> we be accessable by the typeName provided by
+     * featureType.getTypeName().
+     *
+     * @param featureType FetureType to add to DataStore
+     * @throws IOException If featureType cannot be created
+     */
+    @Override
+    public void createSchema(SimpleFeatureType featureType) throws IOException {
+        throw new IllegalAccessError("Modification of schema is not permitted");
+    }
+
+    /**
+     * Used to update a schema in place.
+     *
+     * <p>This functionality is similar to an "alter table" statement in SQL. Implementation is
+     * optional; it may not be supported by all servers or files.
+     *
+     * @param typeName
+     * @param featureType
+     * @throws IOException if the operation failed
+     * @throws UnsupportedOperation if functionality is not available
+     */
+    @Override
+    public void updateSchema(Name typeName, SimpleFeatureType featureType) throws IOException {
+        // do nothing?
```

Comment:
I was going to but one of the earlier iterations couldn't meet the return type of one of the methods - I'll check if that is still an issue because it would be easier